lib/core/src/main/java/com/alsutton/shardedtestexecutor/TestExecutionPredicate.java

```diff
@@ -19,8 +19,8 @@ public TestExecutionPredicate() {
 
     @Override
     public boolean test(String testClassName) {
-        int classHash = Math.abs(classHasher.hash(testClassName));
-        int targetShard = classHash % environment.getTotalShardCount();
+        int classHash = classHasher.hash(testClassName);
+        int targetShard = Integer.remainderUnsigned(classHash, environment.getTotalShardCount());
```

Comment:
An alternative approach is `java.lang.Math#floorMod(int, int)` (thanks to @amaembo in https://twitter.com/tagir_valeev/status/1408344589216399360)

Reply:
Assuming positive shared count, Integer.remainderUnsigned or Math.floorMod have the same overall distribution across all hash codes, but the former may be handled as an intrinsic in some environments (https://twitter.com/jakewharton/status/1311678590715604993, https://bugs.openjdk.java.net/browse/JDK-8238669) while I've never heard of that being proposed for the latter.

Reply:
Yeah, `remainderUnsigned` is probably easier to read (it is clear that the output is unsigned) :)